spectral_cube/stokes_spectral_cube.py

```diff
@@ -0,0 +1,201 @@
+from __future__ import print_function, absolute_import, division
+
+import numpy as np
+
+from astropy.extern import six
+from .spectral_cube import SpectralCube
+from . import wcs_utils
+from .masks import BooleanArrayMask, is_broadcastable_and_smaller
+
+__all__ = ['StokesSpectalCube']
+
+VALID_STOKES = ['I', 'Q', 'U', 'V', 'RR', 'LL', 'RL', 'LR']
+
+
+class StokesSpectralCube(object):
+    """
+    A class to store a spectral cube with multiple Stokes parameters.
+
+    The individual Stokes cubes can share a common mask in addition to having
+    component-specific masks.
```

Comment:
I guess https://github.com/radio-astro-tools/spectral-cube/issues/264 means you want to leave the `Parameters` and other details for a separate PR?